src/plugins/spec/spec.c

```diff
@@ -0,0 +1,842 @@
+/**
+ * @file
+ *
+ * @brief Source for spec plugin
+ *
+ * @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+#include "spec.h"
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <fnmatch.h>
+#include <kdbhelper.h>
+#include <kdbease.h>
+#include <kdberrors.h>
+#include <kdbprivate.h>
+
+#include "metafunctions.h"
+
+#define MAX_CHARS_IN_LONG 26
+
+typedef enum{GET, SET}Direction;
+
+typedef enum{ERROR, WARNING, INFO, IGNORE}OnConflict;
+
+typedef enum{ARRAYMEMBER, INVALID, SUBCOUNT, CONFLICT, OUTOFRANGE, MISSING, NAC}Conflict;
+
+typedef struct
+{
+	OnConflict member;
+	OnConflict invalid;
+	OnConflict count;
+	OnConflict conflict;
+	OnConflict range;
+	OnConflict missing;
+}ConflictHandling;
+
+static char *keyNameToMatchingString(const Key *key)
+{
+	uint8_t arrayCount = 0;
+	char *name = strchr(keyName(key), '/');
+	for(char *ptr = name; *ptr != '\0'; ++ptr)
+		if(*ptr == '#')
+			++arrayCount;
+	char *pattern = elektraMalloc(elektraStrLen(name)+arrayCount);
+	char *dst = pattern;
+	for(char *src = (name+1); *src != '\0'; ++src)
+	{
+		if(*src == '_' && *(src-1) == '/' && (*(src+1) == '/' || *(src+1) == '\0'))
+		{
+			*dst++ = '*';
+		}
+		else if(*src == '#' && *(src-1) == '/' && (*(src+1) == '/' || *(src+1) == '\0'))
+		{
+			*dst++ = '#';
+			*dst++ = '*';
+		}
+		else
+		{
+			*dst++ = *src;
+		}
+	}
+	*dst = '\0';
+	return pattern;
+}
+
+static int matchPatternToKey(const char *pattern, const Key *key)
+{
+	return !fnmatch(pattern, (strchr(keyName(key), '/')+1), FNM_NOESCAPE|FNM_PATHNAME);
```

Comment:
You are right, my example is too simple:
`user/with\\/innerslash/abc.` so a backslash (escaped) before an actual slash (that should separate `with\` and `innerslash`).

In any case, some test cases would be nice.